cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/AbstractRunFixerService.java

```diff
@@ -231,9 +231,16 @@ protected void startUp() throws Exception {
       LOG.debug("Invalid interval specified for the local dataset deleter {}. Setting it to 3600 seconds.", interval);
       interval = 3600L;
     }
-    // Schedule the local dataset deleter with 5 minutes initial delay
+
+    long initialDelay = cConf.getLong(Constants.AppFabric.LOCAL_DATASET_DELETER_INITIAL_DELAY_SECONDS);
+    if (initialDelay <= 0) {
+      LOG.debug("Invalid initial delay specified for the local dataset deleter {}. Setting it to 300 seconds.",
```

Comment:
Since this indicates a misconfiguration, I think this should be more than just DEBUG.